### ISSUE TITLE
IMTLDevice.GetLowPower() sends the incorrect selector

### ISSUE THREAD

**Issue by hamarb123** [xamarin/xamarin-macios#12708]
### Steps to Reproduce

1. Call `IMTLDevice.GetLowPower()`
2. Witness the error `-[BronzeMtlDevice lowPower]: unrecognized selector sent to instance 0x7fe0a01c0000` (or something similar)

Note: GetLowPower is actually an extension method.

### Expected Behavior
According to https://developer.apple.com/documentation/metal/mtldevice/1433409-lowpower?language=objc the `isLowPower` selector should be sent, and we get a boolean back.

### Actual Behavior
The `lowPower` is sent, which doesn't exist therefore it crashes OR it can sometimes just complain in the console.
eg.
```
2021-09-13 10:29:52.123 LowPowerCrashExample[22441:980715] -[MTLIGAccelDevice lowPower]: unrecognized selector sent to instance 0x7fd800290000
2021-09-13 10:29:52.129 LowPowerCrashExample[22441:980715] Failed to set (contentViewController) user defined inspected property on (NSWindow): -[MTLIGAccelDevice lowPower]: unrecognized selector sent to instance 0x7fd800290000
```

### Environment

```
Visual Studio Community 2022 for Mac Preview
Version 17.0 Preview (17.0 build 2706)
Installation UUID: 801b43da-f177-4fe3-808c-cf94b1c8b76b
	GTK+ 2.24.23

Mono Framework MDK
Runtime:
	Mono 6.12.0 (2020-02/eb4c3116ebf) (64-bit)
	Package version: 612000147

Roslyn (Language Service)
3.10.0-3.21226.12+c153357eb74f4b5c8fc905b7ae42168b76515531

NuGet
Version: 6.0.0.129

.NET SDK
SDK: /usr/local/share/dotnet/sdk/6.0.100-rc.2.21456.8/Sdks
SDK Versions:
	6.0.100-rc.2.21456.8
	6.0.100-rc.2.21425.12
	6.0.100-rc.1.21417.3
	6.0.100-rc.1.21403.66
	6.0.100-preview.7.21379.14
	5.0.400
	3.1.412
	3.0.103
	2.2.207
	2.1.817
MSBuild SDKs: /Applications/Visual Studio (Preview).app/Contents/MonoBundle/MSBuild/Current/bin/Sdks

.NET Core Runtime
Runtime: /usr/local/share/dotnet/dotnet
Runtime Versions:
	6.0.0-rc.2.21454.1
	6.0.0-rc.2.21423.6
	6.0.0-rc.1.21415.6
	6.0.0-rc.1.21401.3
	5.0.9
	3.1.18
	3.0.3
	2.2.8
	2.1.29

.NET 5.0 SDK
SDK: 5.0.400

.NET Core 3.1 SDK
SDK: 3.1.412

Xamarin.Profiler
Version: 1.6.15.68
Location: /Applications/Xamarin Profiler.app/Contents/MacOS/Xamarin Profiler

Updater
Version: 11

Xamarin Designer
Version: 17.0.0.58
Hash: 927d548a1
Branch: remotes/origin/d17-0
Build date: 2021-08-27 11:27:00 UTC

Apple Developer Tools
Xcode 13.0 (19223.2)
Build 13A5212g

Xamarin.Mac
Version: 7.14.0.24 (Visual Studio Community)
Hash: c4b89cddb
Branch: d16-10
Build date: 2021-06-15 22:03:00-0400

Xamarin.iOS
Version: 14.20.0.24 (Visual Studio Community)
Hash: c4b89cddb
Branch: d16-10
Build date: 2021-06-15 22:03:01-0400

Xamarin.Android
Version: 11.3.99.54 (Visual Studio Community)
Commit: xamarin-android/main/0e5e06f
Android SDK: /Users/Hamish/Library/Developer/Xamarin/android-sdk-macosx
	Supported Android versions:
		6.0 (API level 23)

SDK Tools Version: 26.1.1
SDK Platform Tools Version: 31.0.2
SDK Build Tools Version: 30.0.2

Build Information: 
Mono: c633fe9
Java.Interop: xamarin/java.interop/main@a5ed891
ProGuard: Guardsquare/proguard/v7.0.1@912d149
SQLite: xamarin/sqlite/3.35.4@85460d3
Xamarin.Android Tools: xamarin/xamarin-android-tools/main@683f375

Microsoft OpenJDK for Mobile
Java SDK: /Users/Hamish/Library/Developer/Xamarin/jdk/microsoft_dist_openjdk_1.8.0.25
1.8.0-25
Android Designer EPL code available here:
https://github.com/xamarin/AndroidDesigner.EPL

Android SDK Manager
Version: 17.0.0.3
Hash: f06a85b
Branch: remotes/origin/HEAD
Build date: 2021-08-27 11:27:02 UTC

Android Device Manager
Version: 17.0.0.2
Hash: d8af29f
Branch: remotes/origin/main
Build date: 2021-08-27 11:27:02 UTC

Build Information
Release ID: 1700002706
Git revision: ba899d0dec8c3161225334edfcc5569ebfae78cc
Build date: 2021-08-27 07:26:10-04
Build branch: release-17.0

Operating System
Mac OS X 11.5.2
Darwin 20.6.0 Darwin Kernel Version 20.6.0
    Wed Jun 23 00:26:31 PDT 2021
    root:xnu-7195.141.2~5/RELEASE_X86_64 x86_64
```

### Build Logs

[msbuild.log.zip](https://github.com/xamarin/xamarin-macios/files/7150592/msbuild.log.zip)

### Example Project (If Possible)

[LowPowerCrashExample.zip](https://github.com/xamarin/xamarin-macios/files/7150597/LowPowerCrashExample.zip)

Note: you'll need a computer that supports metal (not sure if this can be an issue or not).
1. Open the project directory in terminal
2. Run `dotnet build`
3. Run `./bin/Debug/net6.0-macos/osx-x64/LowPowerCrashExample.app/Contents/MacOS/LowPowerCrashExample` (so that you can see the command line output)

**Comment by chamons** [comment]
I can reproduce this on legacy Xamarin.Mac but not in ObjC:

```csharp
            var devs = MTLDevice.GetAllDevices();
            Console.WriteLine(devs.Length);
            Console.WriteLine(devs[0].GetLowPower());
```

```objc
    bool b = [devices[0] isLowPower];
    NSLog (@"%d", b);
    NSLog (@"%d", [devices count]);
```

It is strange that we're getting devices (1) on each, but a declared selector isn't working in ObjC.

Maybe it is code signed related? Some entitlement? Doing more looking.

**Comment by chamons** [comment]
I tried setting up code signing, double checked entitlement on the objc sample (there were none set), tried forcing a -framework Metal, etc.

A bit of a mystery to solve.

**Comment by chamons** [comment]
Manuel figured it out:

    bool b = [devices[0] isLowPower];

vs

the binding which is calling lowPower due to a generator bug.

You can use this manual binding to work around it, as it may be awhile before it is fixed, given the need for a generator change.

```csharp
    static class MetalMonkeyPatch
    {
        [return: System.Runtime.InteropServices.MarshalAs(System.Runtime.InteropServices.UnmanagedType.I1)]
        [System.Runtime.InteropServices.DllImport("/usr/lib/libobjc.dylib", EntryPoint = "objc_msgSend")]
        extern static bool bool_objc_msgSend(IntPtr receiver, IntPtr selector);

        public static bool GetLowPowerValue(this IMTLDevice device)
        {
            return bool_objc_msgSend(device.Handle, ObjCRuntime.Selector.GetHandle("isLowPower"));
        }
    }
```

**Comment by hamarb123** [comment]
@chamons thanks for figuring it out!
Sorry, I should have put my temporary fix there (pretty much the same as yours), it might have helped you guys work out what was going on quicker.
Also though, I did say that it was the wrong selector being sent by .NET.
Thanks again though!

**Comment by rolfbjarne** [comment]
This was fixed in fc14243c179487438717583c048fdc49845a2288.
